addon/addon.go

```diff
@@ -149,20 +152,35 @@ func NewAddonManager(kubeConfig *rest.Config, agentImage string, agentInstallAll
 		return nil, err
 	}
 
+	addonClient, err := addonv1alpha1client.NewForConfig(kubeConfig)
+	if err != nil {
+		return nil, err
+	}
+
 	kubeClient, err := kubernetes.NewForConfig(kubeConfig)
 	if err != nil {
 		klog.Errorf("unable to create kube client: %v", err)
 		return addonMgr, err
 	}
 
 	agentFactory := addonfactory.NewAgentAddonFactory(AppMgrAddonName, ChartFS, ChartDir).
-		WithGetValuesFuncs(getValue, addonfactory.GetValuesFromAddonAnnotation).
+		// register the supported configuration types
+		WithConfigGVRs(
+			schema.GroupVersionResource{Version: "v1", Resource: "configmaps"},
+			schema.GroupVersionResource{Group: "addon.open-cluster-management.io", Version: "v1alpha1", Resource: "addondeploymentconfigs"},
+		).
+		WithGetValuesFuncs(
+			// get the AddOnDeloymentConfig object and transform it to Values object
+			addonfactory.GetAddOnDeloymentConfigValues(
+				addonfactory.NewAddOnDeloymentConfigGetter(addonClient),
+				addonfactory.ToAddOnNodePlacementValues,
+			),
+			getValue,
```

Comment:
we still need `addonfactory.GetValuesFromAddonAnnotation`, we use addon value annotation to configure the proxy and image registry

And I suggest put the `addonfactory.GetAddOnDeloymentConfigValues` at the end of valueFunc slice, thus we can use `AddOnDeloymentConfig` to override other configs